core/src/main/java/org/infinispan/notifications/cachelistener/CacheNotifierImpl.java

```diff
@@ -1467,15 +1414,37 @@ private boolean isClusterListenerAvailable(CacheMode mode) {
                log.tracef("Listener %s initial state for cache completed", generatedId);
             }
 
-            if (t != null) {
-               CompletableFutures.rethrowException(t);
-            }
+            CompletableFutures.rethrowException(t);
```

Comment:
I must admit this looks wrong to me until I look at `rethrowException`. It makes me think it should be a non null exception.

Reply:
Maybe the name could be better, but throwing the exception only if present is the whole point.

If we do the null check outside, then `throw CompletableFutures.asCompletionException(t)` is even clearer and just a few characters more.

Reply:
I renamed the method to `rethrowExceptionIfPresent`, it's not in a `@public` package so I didn't keep the old version.

Reply:
Sounds good.